app/flatpak-cli-transaction.c

```diff
@@ -502,6 +502,50 @@ operation_error (FlatpakTransaction            *transaction,
   return TRUE; /* Continue */
 }
 
+static gboolean
+webflow_start (FlatpakTransaction *transaction,
+               const char         *remote,
+               const char         *url,
+               guint               id)
+{
+  FlatpakCliTransaction *self = FLATPAK_CLI_TRANSACTION (transaction);
+  const char *browser;
+  g_autoptr(GError) local_error = NULL;
+  const char *args[3] = { NULL, url, NULL };
+
+  if (!self->disable_interaction)
+    {
+      g_print (_("Authentication required for remote '%s'\n"), remote);
+      if (!flatpak_yes_no_prompt (TRUE, _("Open browser?")))
+        return FALSE;
+    }
+
+  browser = g_getenv ("BROWSER");
+  if (browser == NULL)
+    browser = "xdg-open";
+
+  /* TODO: Use better way to find default browser */
+
+  args[0] = browser;
+  if (!g_spawn_async (NULL, (char **)args, NULL, G_SPAWN_SEARCH_PATH,
+                      NULL, NULL, NULL, &local_error))
+    {
+      g_printerr ("Failed to spawn browser %s: %s\n", browser, local_error->message);
+      return FALSE;
+    }
+
+  g_print ("Waiting for browser...\n");
+
```

Comment:
Doesn't look like this is actually doing any waiting ?

Reply:
Its waiting for the browser to handle the url, and eventually get redirected to the local host uri telling the Authenticator the token. After that it will send the done signal and continue .

Reply:
As in, it will return to the main loop spinning code.